src/base/utils/fs.cpp

```diff
@@ -234,116 +197,65 @@ bool Utils::Fs::sameFiles(const QString &path1, const QString &path2)
-{
-    QString ret = toUniformPath(filePath);
-    if (ret.endsWith('/'))
-        ret.chop(1);
-    const int slashIndex = ret.lastIndexOf('/');
-    if (slashIndex >= 0)
-    {
-        if (removed)
-            *removed = ret.mid(slashIndex + 1);
-        ret = ret.left(slashIndex);
-    }
-    return ret;
-}
-
-bool Utils::Fs::sameFileNames(const QString &first, const QString &second)
+Path Utils::Fs::toValidPath(const QString &name, const QString &pad)
 {
-#if defined(Q_OS_UNIX) || defined(Q_WS_QWS)
-    return QString::compare(first, second, Qt::CaseSensitive) == 0;
-#else
-    return QString::compare(first, second, Qt::CaseInsensitive) == 0;
-#endif
-}
+    const QRegularExpression regex {QLatin1String("[:?\"*<>|]+")};
 
-QString Utils::Fs::expandPath(const QString &path)
-{
-    const QString ret = path.trimmed();
-    if (ret.isEmpty())
-        return ret;
+    QString validPathStr = name;
+    validPathStr.replace(regex, pad);
 
-    return QDir::cleanPath(ret);
+    return Path(validPathStr);
 }
 
-QString Utils::Fs::expandPathAbs(const QString &path)
+qint64 Utils::Fs::freeDiskSpaceOnPath(const Path &path)
 {
-    return QDir(expandPath(path)).absolutePath();
+    return QStorageInfo(path.toString()).bytesAvailable();
 }
 
-QString Utils::Fs::tempPath()
+Path Utils::Fs::tempPath()
 {
-    static const QString path = QDir::tempPath() + "/.qBittorrent/";
-    QDir().mkdir(path);
+    static const Path path = Path(QDir::tempPath()) / Path(".qBittorrent");
+    mkdir(path);
     return path;
 }
 
-bool Utils::Fs::isRegularFile(const QString &path)
+bool Utils::Fs::isRegularFile(const Path &path)
 {
     struct ::stat st;
-    if (::stat(path.toUtf8().constData(), &st) != 0)
+    if (::stat(path.toString().toUtf8().constData(), &st) != 0)
     {
         //  analyse erno and log the error
         const auto err = errno;
         qDebug("Could not get file stats for path '%s'. Error: %s"
-               , qUtf8Printable(path), qUtf8Printable(strerror(err)));
+               , qUtf8Printable(path.toNative()), qUtf8Printable(strerror(err)));
         return false;
     }
 
     return (st.st_mode & S_IFMT) == S_IFREG;
 }
 
 #if !defined Q_OS_HAIKU
-bool Utils::Fs::isNetworkFileSystem(const QString &path)
+bool Utils::Fs::isNetworkFileSystem(const Path &path)
 {
 #if defined(Q_OS_WIN)
-    const std::wstring pathW {path.toStdWString()};
-    auto volumePath = std::make_unique<wchar_t[]>(path.length() + 1);
-    if (!::GetVolumePathNameW(pathW.c_str(), volumePath.get(), (path.length() + 1)))
+    const std::wstring pathW = path.toNative().toStdWString();
+    auto volumePath = std::make_unique<wchar_t[]>(pathW.length() + 1);
+    if (!::GetVolumePathNameW(pathW.c_str(), volumePath.get(), (pathW.length() + 1)))
         return false;
     return (::GetDriveTypeW(volumePath.get()) == DRIVE_REMOTE);
 #else
-    QString file = path;
+    // NOTE: Why do we need to append "/." to the end of path?
```

Comment:
This should answer you:
https://man7.org/linux/man-pages/man2/fstatfs64.2.html
>path is the pathname of any file within the mounted filesystem.

Therefore it can't be the mount point IIRC (or it can? well you have to test it to be sure...)

Well I don't really expected to see such comments in code, those can be asked on the net instead...

Reply:
> path is the pathname of any file within the mounted filesystem.

But both `some/path/.` and `some/path` point to the same place, don't they?

Reply:
I suppose not, otherwise I would have update it if I found the opportunity.

Anyway, the point is to remove the simplify the following `if()` block if Path won't ever contain trailing slash.

Reply:
> Anyway, the point is to remove the simplify the following `if()` block if Path won't ever contain trailing slash.

Done.